paasta_tools/chronos_tools.py

```diff
@@ -471,23 +471,56 @@ def create_complete_config(service, job_name, soa_dir=DEFAULT_SOA_DIR):
     )
 
     complete_config['name'] = compose_job_id(service, job_name)
+
+    # resolve conflicts between the 'desired_state' and soa_configs disabled
+    # flag.
     desired_state = chronos_job_config.get_desired_state()
+    soa_disabled_state = complete_config['disabled']
+
+    resolved_disabled_state = determine_disabled_state(desired_state,
+                                                       soa_disabled_state)
+
+    if resolved_disabled_state != soa_disabled_state:
+        log.info(("disabled state in deployments.json conflicts with state in"
+                  "soa_configs. setting disabled state to %r" %
+                  resolved_disabled_state))
+
+    complete_config['disabled'] = resolved_disabled_state
 
     # we use the undocumented description field to store a hash of the chronos config.
     # this makes it trivial to compare configs and know when to bounce.
     complete_config['description'] = get_config_hash(complete_config)
 
-    # If the job was previously stopped, we should stop the new job as well
-    # NOTE this clobbers the 'disabled' param specified in the config file!
-    if desired_state == 'start':
-        complete_config['disabled'] = False
-    elif desired_state == 'stop':
-        complete_config['disabled'] = True
-
     log.debug("Complete configuration for instance is: %s" % complete_config)
     return complete_config
 
 
+def determine_disabled_state(branch_state, soa_disabled_state):
+    """ Determine the *real* disabled state. There are two sources of truth for
+    the disabled/enabled state of a job: the 'disabled' flag in soa-configs,
+    and the state set in deployments.json by `paasta stop` or `paasta start`.
+    If the two conflict, then this determines the 'real' state for it.
+
+    We take a defensive stance here: if either source has disabled as True,
```

Comment:
Just throwing this out here.... what if we simply dropped `disabled: ` from soa configs entirely?

Then we have no more need for this function, and the there is only "one" way to enabled/disable a chronos job?